contrib/pylibpod/pylibpod/libs/containers.py

```diff
@@ -0,0 +1,207 @@
+import collections
+import functools
+import json
+
+
+class Container(collections.UserDict):
+    """Model for a container."""
+
+    def __init__(self, client, id, data):
+        """Construct Container Model."""
+        super(Container, self).__init__(data)
+
+        self._client = client
+        self._id = id
+
+        self._refresh(data)
+        assert self._id == self.data['id'],\
+            f'Requested container id({self._id}) does not match store id({self.id})'
+
+    def __getitem__(self, key):
+        """Get items from parent dict + apply aliases."""
+        if key == 'running':
+            key = 'containerrunning'
+        return super().__getitem__(key)
+
+    def _refresh(self, data):
+        super().update(data)
+        for k, v in data.items():
+            setattr(self, k, v)
+        setattr(self, 'running', data['containerrunning'])
+
+    def refresh(self):
+        """Refresh status fields for this container."""
+        ctnr = Containers(self._client).get(self.id)
+        self._refresh(ctnr)
+
+    def attach(self, detach_key=None, no_stdin=False, sig_proxy=True):
+        """Attach to running container."""
+        with self._client() as podman:
+            # TODO: streaming and port magic occur, need arguements
+            podman.AttachToContainer()
+
+    def processes(self):
+        """Show processes running in container."""
+        with self._client() as podman:
+            results = podman.ListContainerProcesses(self.id)
+        for p in results['container']:
+            yield p
+
+    def changes(self):
+        """Retrieve container changes."""
+        with self._client() as podman:
+            results = podman.ListContainerChanges(self.id)
+        return results['container']
+
+    def kill(self, signal=15):
+        """Send signal to container, return id if successful.
+
+        default signal is 15.
+        """
+        with self._client() as podman:
+            results = podman.KillContainer(self.id, signal)
+        return results['container']
+
+    def _lower_hook(self):
+        """Fix keys."""
+
+        @functools.wraps(self._lower_hook)
+        def wrapped(input):
+            return {k.lower(): v for (k, v) in input.items()}
+
+        return wrapped
+
+    def inspect(self):
+        """Retrieve details about containers."""
+        with self._client() as podman:
+            results = podman.InspectContainer(self.id)
+            obj = json.loads(
+                results['container'], object_hook=self._lower_hook())
+            return collections.namedtuple('ContainerInspect', obj.keys())(**obj)
+
+    def export(self, target):
+        """Export container from store to tarball.
+
+        TODO: should there be a compress option, like images?
+        """
+        with self._client() as podman:
+            results = podman.ExportContainer(self.id, target)
+        return results['tarfile']
+
+    def start(self):
+        """Start container, return id on success."""
```

Comment:
@wking This return has waffled, most the of the other functions return ID because it can be used to verify that you are working on the expected container/image.  The opposite of GI-GO.  Here it can just be a warm feeling that what  you asked to be started was.  Vs. podman starting some other random container.